### ISSUE TITLE
Support `consumed_capacity` keyword argument

### ISSUE THREAD

**Issue by GarrettHeel** [mathcamp/flywheel#33]
Functions in `engine.py` should accept and pass along a `consumed_capacity` argument to `dynamo3`

**Comment by stevearc** [comment]
I am actually in the process of updating and rewriting dynamo3 to support all the new dynamo features. Part of that is refactoring the way it handles the returned ConsumedCapacity data. After that's in a better state I'll look adding some of that logic to flywheel.

What's your use case for this? Are you just curious about how much capacity a query takes? Are you trying to do some form of rate-limiting?

**Comment by GarrettHeel** [comment]
Great to hear. Initially I'd like to use the consumed capacity data to collect stats for ad-hoc optimisation. Eventually some kind of rate-limiting or different API behavior could fall out of that.

Will keep an eye on the project, thanks for the quick response!

**Comment by stevearc** [comment]
I just released version `0.4.3` of dynamo3 (with the fix for ConsumedCapacity on BatchGet*). I also added support for setting `connection.default_return_capacity = True`, which will default all of the relevant queries to using `INDEXES` instead of `NONE`. You can also get callbacks for certain events now, like when any call returns a ConsumedCapacity, which I think you can use for your purpose.

```python
conn = DynamoDBConnection.connect('us-west-1')
conn.default_return_capacity = True
def print_capacity(conn, command, query_kwargs, response, capacity):
  print command + ' ' + str(capacity)
conn.subscribe('capacity', print_capacity)
```

I like this solution more than trying to punch through the capacity arguments in the flywheel API, but let me know if there's anything you still need it to do that it can't yet.

**Comment by GarrettHeel** [comment]
That looks great Steve, thanks for the super fast turnaround!
